apps/condo/domains/acquiring/utils/billingCentrifuge.js

```diff
@@ -0,0 +1,374 @@
+            // The rest needed amount for particular recipient
+            let distributionAmount = Big(d.amount)
+                .minus(get(appliedAmounts, recipientKey, Big(0)))
+                .minus(get(appliedFeeAmounts, recipientKey, Big(0)))
+
+            let share = hasEnoughAmountForGroup ? distributionAmount : distributionAmount.div(needToSplitToGroupAmount).times(undistributedAmountAvailableForGroup)
+
+            if (share.lte(0)) {
+                continue
+            }
+
+            const roundedShare = share.round(decimalPlaces, Big.roundHalfUp)
+            restUndistributedAmount = restUndistributedAmount.minus(roundedShare)
+
+            /** @type {TSplit} */
+            let split
+            if (hasEnoughAmountForGroup) {
+                split = {
+                    recipient: d.recipient,
+                    amount: roundedShare.toString(),
+                }
+            } else if (d.vor) {
+                // `hasEnoughAmountForGroup=false` means that this is the last group of recipients to split amount for
+                // Also, this means that VOR-item gets rounding result (the rest undistributed amount)
+
+                // if this is the last recipient within sorted group (vor=true)
+                split = {
+                    recipient: d.recipient,
+                    amount: roundedShare.plus(restUndistributedAmount).toString(),
+                }
+                restUndistributedAmount = Big(0)
+            } else {
+                split = {
+                    recipient: d.recipient,
+                    amount: roundedShare.toString(),
+                }
+            }
+
+            splits.push(split)
+        }
+
+        if (restUndistributedAmount.lte(0)) {
+            break
+        }
+    }
+
+    /** @type {Record<string, TDistribution>} */
+    const distributionsByKey = distribution.reduce((res, d) => {
+        const key = createRecipientKey(d.recipient)
+        res[key] = d
+
+        return res
+    }, {})
+
+    //
+    // Distribute overpayment
+    //
+    let restUndistributedOverpaymentAmount = totalAppliedAmount.plus(paymentAmount).minus(totalDistributionAmount)
+    if (restUndistributedOverpaymentAmount.gt(0)) {
+        const totalOverpaymentAmount = Big(restUndistributedOverpaymentAmount)
+
+        let totalOverpaymentReceiversParts = 0
+        for (const d of distribution) {
+            const key = createRecipientKey(d.recipient)
+            if (distributionsByKey[key].overpaymentPart) {
+                totalOverpaymentReceiversParts += distributionsByKey[key].overpaymentPart
+            }
+        }
+        const distributionsWithOverpayment = distribution
+            .filter((d) => !!d.overpaymentPart)
+            // The victim of the rounding operation MUST be the last item
+            // If there are several vor-item here, sort by order
+            .sort((a, b) => (a.vor && b.vor) ? a.order - b.order : (a.vor ? 1 : -1))
```

Comment:
compare(a, b) === - compare(b, a)

Reply:
this is not correct, where both are not wor

Reply:
The order of not-vor items is not matter here. But ok, I'll sort them :)